### PR TITLE
Add Data Volume annotations documentation

### PR DESCRIPTION
**Pull request by arnongilboa**
Signed-off-by: Arnon Gilboa <agilboa@redhat.com>

**What this PR does / why we need it**:
Explain using specific annotations to a DV or PVC to be passed to the Transfer Pods and control their behavior.

**Special notes for your reviewer**:

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by maya-r** [comment]
Perhaps use this opportunity to explain why only a few annotations are allowed

File: doc/datavolumes.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # Data Volumes
 
 ## Introduction
-Data Volumes(DV) are an abstraction on top of Persistent Volume Claims(PVC) and the Containerized Data Importer(CDI). The DV will monitor and orchestrate the upload/import of the data into the PVC. Once the process is completed, the DV will be in a consistent state that allow consumers to make certain assumptions about the DV in order to progress their own orchestration.
+Data Volumes(DV) are an abstraction on top of Persistent Volume Claims(PVC) and the Containerized Data Importer(CDI). The DV will monitor and orchestrate the upload/import of the data into the PVC. Specific [DV annotations](datavolume-annotation.md) are passed to the transfer pods to control their behavior. Once the process is completed, the DV will be in a consistent state that allow consumers to make certain assumptions about the DV in order to progress their own orchestration.
```
Comment (maya-r) [inline doc/datavolumes.md:4]: I would drop this from the introduction, wanting to pass annotations is a niche use case that wouldn't interest everyone, and we're putting more of a burden to understand what datavolumes are

Comment (arnongilboa) [reply doc/datavolumes.md]: moved down

**Comment by arnongilboa** [comment]
> Perhaps use this opportunity to explain why only a few annotations are allowed

@maya-r added

File: doc/datavolume-annotations.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
+# Data Volume Annotations
+
+## Introduction
+
+CDI allows adding specific annotations to a DV or PVC to be passed to the Transfer Pods and control their behavior. Only specifically allowed annotations are passed, to prevent unexpected behavior due to missing testing coverage. However any needed annotation can be added to the allowed list.
```
Comment (maya-r) [inline doc/datavolume-annotations.md:5]: I assume the allowed list isn't controlled in the "user can change it once running CDI" way.
Not sure that the bit about the allowed list being changeable should be in this documentation.

Comment (arnongilboa) [reply doc/datavolume-annotations.md]: removed

**Pull request merged**